### PR TITLE
Development

### PR DESCRIPTION
**Pull request by FlyingBird95**
Changes:

- Display times as numbers to make them sortable 7ebfef6fe467a8ec3cc021a68f50a7b685f69a5d
- Add leading slash to blueprint paths b3704077fe32b729e04d762b6698045fd249f12d
- Added status codes with corresponding views https://github.com/flask-dashboard/Flask-MonitoringDashboard/commit/a8e84d9009c7c7af4302ed5742b7277e6c58a5a9

### CODE REVIEW

**Comment by FlyingBird95** [comment]
I think the code is ready for master now ;)

@bogdanp05 Do you want to do a last review?
@mircealungu Feel free to also do a review, and accept the PR!!

Patrick

**Comment by bogdanp05** [comment]
Hey Patrick! This release was a long time coming, great to see it here! I'll do a review tonight or tomorrow at the latest:)

File: flask_monitoringdashboard/main.py [hunk 1]
```diff
@@ -19,21 +19,16 @@
...
-    time.sleep(2)
-    time.sleep(1)
-
-
-@app.route('/endpoint')
+@app.route('/endpointd')
 def endpoint():
-    f()
-    return 'Ok'
+    print("Hello, world")
```
Comment (bogdanp05) [inline flask_monitoringdashboard/main.py]: Hello, FMD!:)

File: flask_monitoringdashboard/main.py [hunk 2]
```diff
@@ -51,11 +46,16 @@ def endpoint4():
...
+    time.sleep(0.2)
+    return 'Ok'
+
+
 def my_func():
     # here should be something actually useful
     return 33.3
 
-
 # schedule = {'weeks': 0,
```
Comment (bogdanp05) [inline flask_monitoringdashboard/main.py]: we should remove it in the next commit

File: flask_monitoringdashboard/static/js/filters.js [hunk 3]
```diff
@@ -1,5 +1,11 @@
...
 
+app.filter('duration_ms', function () {
+    return function (time) {
+        return Math.round(parseFloat(time) * 10) / 10;
+    }
+});
+
 app.filter('duration', function () {
     return function (time) {
         // ms is a float or int
```
Comment (bogdanp05) [inline flask_monitoringdashboard/static/js/filters.js]: Remove line 12

File: flask_monitoringdashboard/controllers/requests.py [hunk 4]
```diff
@@ -30,6 +35,74 @@ def get_num_requests_data(db_session, start_date, end_date):
...
+def get_error_requests(db_session, endpoint_id, *criterion):
+    """
+    Gets all requests that did not return a 200 status code.
+
+    :param db_session:
+    :param endpoint_id:
+    :param criterion:
+    :return:
+    """
+    return db_session \
```
Comment (mircealungu) [inline flask_monitoringdashboard/controllers/requests.py]: @thejager it's very likely you knew, but you can split something over multiple lines if you put the expression between parentheses: that way you don't have to terminate lines with \

File: flask_monitoringdashboard/main.py [hunk 5]
```diff
@@ -19,30 +19,32 @@
...
 @app.route('/endpoint3')
 def endpoint3():
-    if random.randint(0, 1) == 0:
-        time.sleep(0.1)
-    else:
-        time.sleep(0.2)
-    return 'Ok'
+    return 'Nice', 203
+    #
+    # if random.randint(0, 1) == 0:
```
Comment (mircealungu) [inline flask_monitoringdashboard/main.py]: why is this commented code in here?

File: flask_monitoringdashboard/static/js/services/menu.js [hunk 6]
```diff
@@ -12,16 +12,20 @@ app.service('menuService', function ($http, endpointService) {
...
 
-        if (page === 'endpoint_hourly' || page === 'endpoint_user_version' || page === 'endpoint_ip' ||
-            page === 'endpoint_version' || page === 'endpoint_user' || page === 'endpoint_profiler' ||
-            page === 'endpoint_grouped_profiler' || page === 'endpoint_outlier') {
+        var endpointPages = [
+            'endpoint_hourly', 'endpoint_user_version', 'endpoint_ip', 'endpoint_version', 'endpoint_user',
+            'endpoint_profiler', 'endpoint_grouped_profiler', 'endpoint_outlier', 'status_code_distribution'
+        ];
+
+        if (endpointPages.includes(page)) {
```
Comment (mircealungu) [inline flask_monitoringdashboard/static/js/services/menu.js]: nice :)

File: flask_monitoringdashboard/controllers/requests.py (hunk 4 above)
Comment (thejager) [reply flask_monitoringdashboard/controllers/requests.py]: I didn't know actually, will change this soon 😎

**Pull request merged**